runtime/lua/vim/inspector.lua

```diff
@@ -0,0 +1,207 @@
+---@field extmarks boolean include extmarks (defaults to true)
+---@field extmarks_with_hl_group boolean when false, also include extmarks without a hl_group (defaults to true)
+---@field semantic_tokens boolean include semantic tokens (defaults to true)
+local defaults = {
+  syntax = true,
+  treesitter = true,
+  extmarks = true,
+  extmarks_with_hl_group = true,
+  semantic_tokens = true,
+}
+
+---Get all the items at a given buffer position
+---@param buf? buffer defaults to the current buffer
+---@param row? number 0-indexed row. Defaults to the current cursor
+---@param col? number 0-indexed col. Defaults to the current cursor
+---@param filter? InspectorFilter (table|nil) a table with key-value pairs to filter the items
+---               - syntax (boolean): include syntax based highlight groups (defaults to true)
+---               - treesitter (boolean): include treesitter based highlight groups (defaults to true)
+---               - extmarks (boolean): include extmarks (defaults to true)
+---               - extmarks_with_hl_group (boolean): when false, also include extmarks without a hl_group (defaults to true)
+---               - semantic_tokens (boolean): include semantic tokens (defaults to true)
+function M.get(buf, row, col, filter)
+  filter = vim.tbl_deep_extend('force', defaults, filter or {})
+
+  buf = buf or 0
+  if row == nil or col == nil then
+    -- get the row/col from the first window displaying the buffer
+    local win = buf == 0 and vim.api.nvim_get_current_win() or vim.fn.bufwinid(buf)
+    if win == -1 then
+      error('row/col is required for buffer not visible in a window')
+    end
+    local cursor = vim.api.nvim_win_get_cursor(win)
+    row, col = cursor[1] - 1, cursor[2]
+  end
+  buf = buf == 0 and vim.api.nvim_get_current_buf() or buf
+
+  local results = {
+    treesitter = {},
+    syntax = {},
+    extmarks = {},
+    semantic_tokens = {},
+  }
+
+  -- resolve hl links
+  ---@private
+  local function resolve_hl(data)
+    if data.hl_group then
+      local hlid = vim.api.nvim_get_hl_id_by_name(data.hl_group)
+      local name = vim.fn.synIDattr(vim.fn.synIDtrans(hlid), 'name')
+      data.hl_group_link = name
+    end
+    return data
+  end
+
+  -- treesitter
+  if filter.treesitter then
+    for _, capture in pairs(vim.treesitter.get_captures_at_pos(buf, row, col)) do
+      capture.hl_group = '@' .. capture.capture
+      table.insert(results.treesitter, resolve_hl(capture))
+    end
+  end
+
+  -- syntax
+  if filter.syntax then
+    for _, i1 in ipairs(vim.fn.synstack(row + 1, col + 1)) do
+      table.insert(results.syntax, resolve_hl({ hl_group = vim.fn.synIDattr(i1, 'name') }))
+    end
+  end
+
+  -- semantic tokens
+  if filter.semantic_tokens then
+    for _, token in ipairs(vim.lsp.semantic_tokens.get_at_pos(buf, row, col) or {}) do
+      token.hl_group = '@' .. token.type
+      table.insert(results.semantic_tokens, resolve_hl(token))
```

Comment:
Any chance you can add modifiers here as well? The modifier interface isn't finalized yet but it could be useful to see what's there for now.